core/src/main/java/com/onelogin/saml2/settings/Saml2Settings.java

```diff
@@ -911,12 +930,8 @@ private boolean checkIdpx509certRequired () {
 			LOGGER.error(errorMsg);
 		}
 
-		if ((this.getAuthnRequestsSigned() == true ||
-			  this.getLogoutRequestSigned() == true ||
-			  this.getLogoutResponseSigned() == true ||
-			  this.getWantAssertionsEncrypted() == true ||
-			  this.getWantNameIdEncrypted() == true)
-			  && this.checkSPCerts() == false) {
+		if (this.getHsm() == null && (this.getAuthnRequestsSigned() || this.getLogoutRequestSigned()
```

Comment:
Replace by:
```
if ((this.getAuthnRequestsSigned() == true ||
			  this.getLogoutRequestSigned() == true ||
			  this.getLogoutResponseSigned() == true ||
			  this.getWantAssertionsEncrypted() == true ||
			  this.getWantNameIdEncrypted() == true)
			  && !this.checkSPCerts() && this.getHsm() == null) {
```

Reply:
Hi @pitbulk 

First of all, sorry for my late reply! I have committed the requested changes on my forked repository. Shall you re-open this pull request or shall I open a new one in order to merge these changes?